src/ios/getHeaderSearchPath.js

```diff
@@ -0,0 +1,38 @@
+const path = require('path');
+const union = require('lodash.union');
+
+/**
+ * Returns last item from an array
+ */
+const lastItem = (array) => array[array.length - 1];
+
+/**
+ * Given an array of headers it returns search path so Xcode can find them
+ * If there are multiple header files located accross different folders,
+ * it finds the top-most one and returns recursive version of it
+ */
+module.exports = function getHeaderSearchPath(sourceDir, headers) {
```

Comment:
Probably it may be divided for two functions: search for a directory inside `headers` argument (it seems name `files` sounds more valuable for this argument) and finding common directory for all these `files` (`headers`)